tests/test_db_runtime_security.py

```
@@ -144,914 +173,937 @@ def _connect_side_effect(*args, **kwargs):
+    """Test write context manager rolls back on error."""
+    manager, _ = db_manager
+    # Create initial table
+    with manager.write() as cursor:
+        cursor.execute(TEST_SQL_CREATE_TABLE)
+        cursor.execute(TEST_SQL_INSERT_VALUE, ("initial",))
+
+    # Attempt operation that will fail
+    with pytest.raises(sqlite3.IntegrityError):
+        with manager.write() as cursor:
+            cursor.execute("INSERT INTO test (id, value) VALUES (1, ?)", ("conflict",))
+            cursor.execute("INSERT INTO test (id, value) VALUES (1, ?)", ("conflict",))
```

Ruff (0.15.12)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)